src/Core/Services/ApiService.cs

```diff
@@ -790,12 +798,99 @@ public async Task<string> GetUsernameFromAsync(ForwardedEmailServiceType service
+            }
+            else if (result["methodResponses"][0][0].ToString() == "error")
+            {
+                throw new Exception("Fastmail error: " + result["methodResponses"][0][1]?["description"].ToString());
+            }
+            throw new Exception("Fastmail error: could not parse response.");
+        }
+
+        private async Task<string> CreateFastmailRequest(string apiKey)
+        {
+            using (var httpclient = new HttpClient())
+            {
+                HttpResponseMessage response;
+                try
+                {
+                    httpclient.DefaultRequestHeaders.Add("Authorization", $"Bearer {apiKey}");
+                    httpclient.DefaultRequestHeaders.Add("Accept", "application/json");
+                    response = await httpclient.GetAsync(new Uri("https://api.fastmail.com/jmap/session"));
+                }
+                catch (Exception e)
+                {
+                    throw new ApiException(HandleWebError(e));
+                }
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new ApiException(new ErrorResponse
+                    {
+                        StatusCode = response.StatusCode,
+                        Message = $"Fastmail error: {(int)response.StatusCode} {response.ReasonPhrase}."
+                    });
+                }
+                var result = JObject.Parse(await response.Content.ReadAsStringAsync());
+                var accountId = result["primaryAccounts"]?["https://www.fastmail.com/dev/maskedemail"]?.ToString();
+                var requestJObj = new JObject
+                {
+                    new JProperty("using",
+                        new JArray { "https://www.fastmail.com/dev/maskedemail", "urn:ietf:params:jmap:core" }),
+                    new JProperty("methodCalls",
+                        new JArray
+                        {
+                            new JArray
+                            {
+                                "MaskedEmail/set",
+                                new JObject
+                                {
+                                    ["accountId"] = accountId,
+                                    ["create"] = new JObject
+                                    {
+                                        ["new-masked-email"] = new JObject
+                                        {
+                                            ["state"] = "enabled",
+                                            ["description"] = "",
+                                            ["url"] = "",
+                                            ["emailPrefix"] = ""
+                                        }
+                                    }
+                                },
+                                "0"
+                            }
+                        })
+                };
+                return requestJObj.ToString();
+            }
+        }
```

Comment:
The `ApiService` is gigantic and doing a lot of stuff so please move this to a new service specific for Fastmail: `FastmailService` and extract the repeated hardcoded values into constants, like `methodResponses`.
Also, would it be a good path when `accountId` is `null`? I mean, is it worth it to send the request or should we just abort/throw an exception?
Furthermore, it'd be good to have a specific exception for the fastmail instead of just `throw new Exception(...)` so if it get's logged we can identify it easily by its type, e.g. `FastmailException`